### PR TITLE
Internally use estimate_gas for settlement simulation at current block

### PR DESCRIPTION
**Pull request by vkgnosis**
Estimate gas internally simulates the transaction which tells us whether
the simulation succeeded and how much gas was used in one call.

Next I will change the driver to use this function so that we do not simulate twice.

### Test Plan
Adapted the ignored test to call the new function and ran the test.

### CODE REVIEW

**Comment by vkgnosis** [comment]
Draft while I debug an issue.

File: solver/src/settlement_simulation.rs
Comment (josojo) [inline solver/src/settlement_simulation.rs:99]: This function name seems quite unspecific, maybe get_settle_method() would be a better name?

**Review by josojo — APPROVED** [review]
Nice, LGTM

File: solver/src/driver.rs
Comment (nlordell) [inline solver/src/driver.rs:234]: One advantage of `Ok(())` is that if the type of the `result` changes, you would get a compiler error to update at this location. Very minor though, and I also see the argument that `Ok(_)` could more clearly communicate the intent of "it succeeded, don't care about the result".

Not requesting any changes, just rambling.

File: solver/src/driver.rs
Comment (nlordell) [inline solver/src/driver.rs:251]: naming nit:
```suggestion
            match settlement_simulation::simulate_and_error_with_tenderly_link(
```

**Review by nlordell — APPROVED** [review]
:pray: Looks good!

Approving assuming stray `dbg!(_)` will get cleaned up before merging.

File: solver/src/driver.rs
Comment (vkgnosis) [reply solver/src/driver.rs:234]: This is exactly why it's _ now. The type has changed from () to U256 for the gas estimation. Next PR will make use of the value.

File: solver/src/settlement_simulation.rs
Comment (nlordell) [reply solver/src/settlement_simulation.rs:166]: Oh this is test code :man_facepalming:

**Pull request merged**
